src/tasks/familiarweight.ts

```diff
@@ -109,14 +110,19 @@ export const FamiliarWeightQuest: Quest = {
             ),
             $item`love song of icy revenge`
           );
+        const heaviestWeight = familiarWeight(chooseHeaviestFamiliar()) + (have($item`astral pet sweater`) ? 10 : 0);
+        const commaWeight = 6 + 11 * get("homemadeRobotUpgrades");
         if (
           have($skill`Summon Clip Art`) &&
           !get("instant_saveClipArt", false) &&
-          ($familiars`Mini-Trainbot, Exotic Parrot`.some((fam) => have(fam)) ||
-            $familiars`Comma Chameleon, Homemade Robot`.every((fam) => have(fam)))
+          ((have($familiar`Mini-Trainbot`) && ((familiarWeight($familiar`Mini-Trainbot`) + 25) > heaviestWeight)) ||
+          (have($familiar`Exotic Parrot`) && ((familiarWeight($familiar`Exotic Parrot`) + 15) > heaviestWeight)) ||
```

Comment:
If the `heaviestWeight` check is included in this conditional, we should also probably include in below when we are deciding between using the trainbot and parrot (if the comma conditional is not satisfied).

Not a big deal; I'll probably do some refactoring after we get this merged in so I can handle that later, unless you can get around to it too. Specifically I don't really like the structure
```
if (condA or condB)
{
  if (condA)
  {
    // stuff
  }
  else if (condB)
  {
    // other stuff
  }
}
```
especially if our conditionals are growing longer and longer.

It should probably be refactored to something like
```
if (usingClipArt)
{
  if (usingComma)
  {
    // Do comma stuff
  }
  else if (usingTrainbot)
  {
    // Do trainbot stuff
  }
  else if (usingParrot)
  {
    // Do parrot stuff
  }
}
```
and this also cleanly resolves the issue of the different trainbot/parrot conditionals

Reply:
I considered if they were currently needed in the other checks and they weren't. But I also prefer moving conditionals to variables so have done so.